### PR TITLE
[wip] feat: promisify app.importCertificate

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Promisifies `app.importCertificate` as part of the promisification initiative.

/cc @MarshallOfSound @ckerr @deepak1556 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

#### Release Notes

Notes: promisifies app.importCertificate()

### CODE REVIEW

File: docs/api/promisification.md [hunk 1]
```diff
@@ -8,57 +8,64 @@ When a majority of affected functions are migrated, this flag will be enabled by
...
-- [ ] [contents.savePage(fullPath, saveType, callback)](https://github.com/electron/electron/blob/master/docs/api/web-contents.md#savePage)
-- [ ] [webFrame.executeJavaScript(code[, userGesture, callback])](https://github.com/electron/electron/blob/master/docs/api/web-frame.md#executeJavaScript)
-- [ ] [webFrame.executeJavaScriptInIsolatedWorld(worldId, scripts[, userGesture, callback])](https://github.com/electron/electron/blob/master/docs/api/web-frame.md#executeJavaScriptInIsolatedWorld)
-- [ ] [webviewTag.executeJavaScript(code[, userGesture, callback])](https://github.com/electron/electron/blob/master/docs/api/webview-tag.md#executeJavaScript)
-- [ ] [webviewTag.printToPDF(options, callback)](https://github.com/electron/electron/blob/master/docs/api/webview-tag.md#printToPDF)
-- [ ] [webviewTag.getZoomFactor(callback)](https://github.com/electron/electron/blob/master/docs/api/webview-tag.md#getZoomFactor)
-- [ ] [webviewTag.getZoomLevel(callback)](https://github.com/electron/electron/blob/master/docs/api/webview-tag.md#getZoomLevel)
+- [win.hookWindowMessage(message, callback)](https://github.com/electron/electron/blob/master/docs/api/browser-window.md#hookWindowMessage)
```
Comment (miniak) [inline docs/api/promisification.md]: let's not change the format in this PR, I will create a separate PR for that to keep the change smaller here

Comment (miniak) [reply docs/api/promisification.md]: https://github.com/electron/electron/pull/16434

File: atom/browser/api/atom_api_app.cc [hunk 2]
```diff
@@ -1040,31 +1042,34 @@ Browser::LoginItemSettings App::GetLoginItemSettings(mate::Arguments* args) {
...
-  callback.Run(rv);
+  promise->Resolve(
+      ImportIntoCertStore(certificate_manager_model_.get(), options));
+  return promise->GetHandle();
 }
 
 void App::OnCertificateManagerModelCreated(
     std::unique_ptr<base::DictionaryValue> options,
-    const net::CompletionCallback& callback,
+    scoped_refptr<util::Promise> promise,
```
Comment (shahzadlone) [inline atom/browser/api/atom_api_app.cc]: Should this perhaps be a referance?

Comment (codebytere) [reply atom/browser/api/atom_api_app.cc]: no, you'll see if you look around that this is how we've implemented all other promisification instances.

**Pull request closed without merging**